src/Spec2-Layout/SpExecutableLayout.class.st

```diff
@@ -343,6 +341,7 @@ SpExecutableLayout >> whenChildrenRemovedDo: aBlock [
 	self announcer
 		when: SpChildrenRemoved
 		do: aBlock
+		for: aBlock receiver
```

Comment:
"aBlock receiver" is not good here...  for clean blocks it will be nil

Reply:
oops I resolved by mistake :P